tests/models/test_renderedtifields.py

```diff
@@ -158,16 +163,35 @@ def test_get_templated_fields(self, templated_field, expected_rendered_field, da
         assert ti.dag_id == rtif.dag_id
         assert ti.task_id == rtif.task_id
         assert ti.run_id == rtif.run_id
-        assert expected_rendered_field == rtif.rendered_fields.get("bash_command")
+        if type(templated_field) is set:
+            # the output order of a set is non-deterministic and can change per process.
+            # this validation can fail if that happens before stringification, so we convert to set and compare.
+            assert ast.literal_eval(expected_rendered_field) == ast.literal_eval(
+                rtif.rendered_fields.get("bash_command")
+            )
+        else:
+            assert expected_rendered_field == rtif.rendered_fields.get("bash_command")
 
         session.add(rtif)
         session.flush()
 
-        assert RTIF.get_templated_fields(ti=ti, session=session) == {
-            "bash_command": expected_rendered_field,
-            "env": None,
-            "cwd": None,
-        }
+        if type(templated_field) is set:
```

Comment:
```suggestion
        if isinstance(templated_field, set):
```

Better from OOPs perspective

Reply:
This is in tests we control the parameters so not a big deal here anyway and more a "nit" -- but to illustrate the reason:

```py
In [13]: class CustomSet(set):
    ...:     def add(self, element):
    ...:         print(f"Adding element: {element}")
    ...:         super().add(element)
    ...:

In [15]: a = CustomSet([1,2])

In [16]: isinstance(a, set)
Out[16]: True

In [17]: type(a) is set
Out[17]: False
```

Reply:
Actually I have a better idea to not have to deal with this problem, let me push a change

Reply:
Nice! Thanks. Let me commit that